hexa/analytics/views.py

```diff
@@ -0,0 +1,30 @@
+import json
+
+from django.http import HttpRequest, HttpResponse, JsonResponse
+from django.views.decorators.csrf import csrf_exempt
+from django.views.decorators.http import require_POST
+
+from hexa.user_management.models import AnonymousUser
+
+from .api import track
+
+
+@require_POST
+@csrf_exempt
+def track_event(request: HttpRequest) -> HttpResponse:
+    """This API endpoint is called by the frontend to track events."""
+    payload = json.loads(request.body.decode("utf-8"))
+
+    if "event" not in payload:
+        return JsonResponse({"Bad request": "event name is required."}, status=400)
+
+    if isinstance(request.user, AnonymousUser) or request.user.analytics_enabled:
```

Comment:
I think it would make sense to check this condition before loading the json.

Reply:
Well, this condition is already checked in the track() in fact...

Reply:
Yes,  I thought it would be better to let the user knows the reason why. WDYT?